test/modules/encoders/test_mdetr_text_encoder.py

```diff
@@ -0,0 +1,198 @@
+                0.71950,
+            ]
+        )
+        out = self.embeddings(self.input_ids)
+        actual = out[1, :, 1]
+        self.assertEqual(
+            out.size(), (self.batch_size, self.input_length, self.hidden_size)
+        )
+        assert_expected(actual, expected, rtol=0.0, atol=1e-4)
+
+    def test_mdetr_wrapped_transformer(self):
+        inp = torch.rand((2, 16, 768))
+        self.inp = inp
+        expected = torch.Tensor(
+            [
+                -0.3351,
+                0.0728,
+                -0.2649,
+                -0.0369,
+                -0.2435,
+                -0.3580,
+                -0.1809,
+                -0.2042,
+                -0.0316,
+                -0.3408,
+                0.0784,
+                -0.1908,
+                -0.2906,
+                -0.0650,
+                -0.3023,
+                -0.1622,
+            ]
+        )
+        out = self.wrapped_transformer_encoder(inp, self.attention_mask)
+        actual = out[1, :, 1]
+        self.assertEqual(
+            out.size(), (self.batch_size, self.input_length, self.hidden_size)
+        )
+        assert_expected(actual, expected, rtol=0.0, atol=1e-4)
+
+    def test_mdetr_text_encoder(self):
+        expected = torch.Tensor(
+            [
+                0.04690,
+                0.06246,
+                0.05839,
+                0.05727,
+                0.06413,
+                0.05640,
+                0.06599,
+                0.04888,
+                0.07137,
+                0.05381,
+                0.06141,
+                0.06225,
+                0.04788,
+                0.06256,
+                0.05058,
+                0.06433,
+            ]
+        )
+        out = self.text_encoder(self.input_ids, self.attention_mask)
+        actual = out[1, :, 1]
+        self.assertEqual(
+            out.size(), (self.batch_size, self.input_length, self.hidden_size)
+        )
+        assert_expected(actual, expected, rtol=0.0, atol=1e-4)
+
+    # Unlike with the encoder where all weights are already present in the TorchText checkpoint,
+    # for the embedding we need to construct our own weights since word_embeddings are not present
+    # in the TorchText model bundle.
+    def _populate_embedding_weights(self):
+        embeddings_state_dict = {}
+        for k, v in self.embeddings.state_dict().items():
+            # These have already been set on init and shouldn't be changed
+            if k == "position_ids":
+                embeddings_state_dict[k] = v
+            # Set these to match the HF RoBERTa weights
+            elif k == "token_type_embeddings.weight":
+                embeddings_state_dict[k] = torch.zeros(v.size())
+            # Set all other weights randomly
+            else:
+                embeddings_state_dict[k] = torch.rand(v.size())
+        self.embeddings.load_state_dict(embeddings_state_dict)
+
+    def _populate_wrapped_transformer_weights(self):
```

Comment:
this test seems more complex that needed . why use the wrapped terminology and loading actual roberta weights?

Reply:
Yeah the idea here was to explicitly check that the two pretrained checkpoints line up, since that is what had caused so much trouble previously. Agree it can be less complex though. Let me address this in a follow-up if that's good with you